internal/policy/operator/deployable_by_olm.go

```diff
@@ -248,13 +251,22 @@ func (p *DeployableByOlmCheck) operatorMetadata(ctx context.Context, bundleRef i
 		installModes[val.Type] = val
 	}
 
+	// validating that package name complies with DNS-1035 labeling constraints
+	// ensuring that CatalogSources can be created/referenced in cluster and reconciles properly
+	// if there is an error we need to prefix the name so the install/reconciles can continue
+	appName := packageName
+	if msgs := validation.IsDNS1035Label(packageName); len(msgs) != 0 {
+		logger.V(log.DBG).Info(fmt.Sprintf("package name %s does not comply with DNS-1035, prefixing to avoid errors", packageName))
+		appName = "p-" + packageName
```

Comment:
This still needs to be re-run through the DNS label validation. Prefixing the value just fixes one possible invalid scenario.

Reply:
@komish Are you sure? In my testing, and our tests, this fixes both issues that are thrown. Also, if we re-validate, and it fails, then what do we do?

Reply:
What if the package is called `1app_package`

Reply:
I suppose this is non-blocking. You're addressing a specific failure that exists, and what I'm asking is a potential (but non-existent) future state. We already have an example in the wild that needs this workaround, but we haven't had an example of what I'm asking you to do yet, and it's unlikely we would given certification would fail if the CatalogSource doesn't come up.

LGTM given

Reply:
I think if this is the case, we'd want this to fail, since that name probably can't be used in other processes. Right now, we really are just trying to solve for the one case for `3scale`.